sqlserver/datadog_checks/sqlserver/const.py

```diff
@@ -268,3 +268,73 @@
 ]
 
 PROC_CHAR_LIMIT = 500
+
+# Schemas
+DEFAULT_SCHEMAS_COLLECTION_INTERVAL = 1200
+
+DB_QUERY = """
+SELECT
+    db.database_id AS id, db.name AS name, db.collation_name AS collation, dp.name AS owner
+FROM
+    sys.databases db LEFT JOIN sys.database_principals dp ON db.owner_sid = dp.sid
+WHERE db.name = '{}';
+"""
+
+SCHEMA_QUERY = """
+SELECT
+    s.name AS name, s.schema_id AS id, dp.name AS owner_name
+FROM
+    sys.schemas AS s JOIN sys.database_principals dp ON s.principal_id = dp.principal_id
+WHERE s.name NOT IN ('sys', 'information_schema')
+"""
+
+TABLES_IN_SCHEMA_QUERY = """
+SELECT
+    object_id AS id, name
+FROM
+    sys.tables
+WHERE schema_id={}
```

Comment:
We are selecting all tables (and also partitions, constraints, etc.) with each check run. Is it viable to select and emit only objects whose definitions changed since the last successful run? [sys.objects.modify_date](https://learn.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-objects-transact-sql?view=sql-server-ver16) contains the timestamp of the last change. That would be a massive optimization making it feasible to handle databases containing hundreds thousands of objects.

Reply:
that's true but we discussed with Justin that it can be a next step. Same for Postgres.

It is not as trivial as it seems - 

1. there are cases where we stop after ingesting some columns. Which means we cannot just query for all changed objects but we need to know what has been ingested in the previous step. 
2. message loss - do we do something for that or we expect messages never to be lost ? for example if backend receives a diff for a table that it doesn't have shell it output an error msg , it cannot communicate back to the agent ?
3. if schema/table is deleted I am not sure it is reflected in modify_date as object disappears.
4. more backend work

There is another optimisation which(orthogonal) may be needs to be done first - probably we can decide to split work between checks first to be able to ingest more data incrementally.